crates/fuel-core/src/graphql_api/ports.rs

```diff
@@ -108,6 +113,15 @@ pub trait OffChainDatabase: Send + Sync {
         direction: IterDirection,
     ) -> BoxedIter<StorageResult<(TxPointer, TxId)>>;
 
+    fn coins_to_spend(
+        &self,
+        owner: &Address,
+        asset_id: &AssetId,
+        target_amount: u64,
+        max_coins: u16,
+        excluded_ids: &ExcludedKeysAsBytes,
```

Comment:
Currently we do not have limits on GraphQL side. And both legacy and new implementation iterate over `excluded_ids` so indeed this could potentially be problematic.

I'll added the limit, equal to max inputs in this commit: a5c80e171d62098e942cbee4b7440a0344222359

User will be presented with the following error:
![image](https://github.com/user-attachments/assets/01bcaac1-ef72-4b69-b8f8-5a4c40406cdb)